test-d/exact.ts

```diff
@@ -143,3 +143,69 @@ import type {Exact} from '../index';
 		fn(input);
 	}
 }
+
+{ // Spec - union of array
+	type Type = Array<{x: string}> & Array<{z: number}>;
+	const fn = <T extends Exact<Type, T>>(args: T) => args;
+
+	{ // It should accept valid input
+		const input = [{
+			x: '',
+			z: 1,
+		}];
+		fn(input);
+	}
+
+	{ // It should reject missing field
+		const input = [{
+			z: 1,
+		}];
+		// @ts-expect-error
+		fn(input);
+	}
+
+	{ // It should reject missing field
+		const input = [{
+			x: '',
+		}];
+		// @ts-expect-error
+		fn(input);
+	}
+
+	{ // It should reject incorrect type
+		const input = [{
+			x: 1,
+			z: 1,
+		}];
+		// @ts-expect-error
+		fn(input);
+	}
+
+	{ // It should reject excess field
+		const input = [{
+			x: '',
+			y: '',
+			z: 1,
+		}];
+		// @ts-expect-error
+		fn(input);
+	}
+}
+
+{ // Spec - union of array with nested fields
+	type Type = Array<{x: string}> & Array<{z: number; d: {e: string}}>;
+	const fn = <T extends Exact<Type, T>>(args: T) => args;
+
+	{ // It should reject excess field
+		const input = [{
+			x: '',
+			z: 1,
+			d: {
+				e: 'test',
+				f: true, // Excess field
+			},
+		}];
+		// @ts-expect-error
+		fn(input);
+	}
```

Comment:
Possibly we can include more test coverages to check excess, exact and missing fields for the union array test